x-pack/legacy/plugins/apm/public/components/shared/TransactionActionMenu/sections.ts

```diff
@@ -0,0 +1,268 @@
+          time,
+          filter: `trace.id:"${transaction.trace.id}" OR ${transaction.trace.id}`
+        }
+      }),
+      condition: true
+    }
+  ];
+
+  const kibanaActions: Action[] = [
+    {
+      label: i18n.translate(
+        'xpack.apm.transactionActionMenu.viewSampleDocumentLinkLabel',
+        {
+          defaultMessage: 'View sample document'
+        }
+      ),
+      href: getDiscoveryHref({
+        basePath,
+        query: getDiscoverQuery(transaction),
+        location
+      }),
+      condition: true
+    },
+    {
+      label: i18n.translate('xpack.apm.transactionActionMenu.viewInUptime', {
+        defaultMessage: 'View monitor status'
+      }),
+      href: uptimeLink,
+      condition: !!transaction.url?.domain
+    }
+  ];
+
+  const sections: Section = {
+    observability: [
+      {
+        title: i18n.translate('xpack.apm.transactionActionMenu.pod.title', {
+          defaultMessage: 'Pod details'
+        }),
+        subtitle: i18n.translate(
+          'xpack.apm.transactionActionMenu.pod.subtitle',
+          {
+            defaultMessage:
+              'View logs and metrics of this pod to get further details.'
+          }
+        ),
+        actions: podActions
+      },
+      {
+        title: i18n.translate(
+          'xpack.apm.transactionActionMenu.container.title',
+          {
+            defaultMessage: 'Container details'
+          }
+        ),
+        subtitle: i18n.translate(
+          'xpack.apm.transactionActionMenu.container.subtitle',
+          {
+            defaultMessage:
+              'View logs and metrics of this container to get further details.'
+          }
+        ),
+        actions: containerActions
+      },
+      {
+        title: i18n.translate('xpack.apm.transactionActionMenu.host.title', {
+          defaultMessage: 'Host details'
+        }),
+        subtitle: i18n.translate(
+          'xpack.apm.transactionActionMenu.host.subtitle',
+          {
+            defaultMessage: 'View host logs to get further details.'
+          }
+        ),
+        actions: hostActions
+      },
+      {
+        title: i18n.translate('xpack.apm.transactionActionMenu.trace.title', {
+          defaultMessage: 'Trace details'
+        }),
+        subtitle: i18n.translate(
+          'xpack.apm.transactionActionMenu.trace.subtitle',
+          {
+            defaultMessage: 'View the trace logs to get further details.'
+          }
+        ),
+        actions: traceActions
+      }
+    ],
+    kibana: [{ actions: kibanaActions }]
+  };
+
+  // Filter out actions that shouldnt be shown and sections without any actions.
+  return Object.values(sections).map(items =>
+    items
+      .reduce((acc: SectionItem[], curr: SectionItem) => {
+        const actions = curr.actions.filter(
+          (action: Action) => action.condition
+        );
+        acc.push({ ...curr, actions });
+        return acc;
+      }, [])
+      .filter(section => section.actions.length > 0)
```

Comment:
Is this really filtering out sections without actions? Or is `section` actually a `sectionItem`?

Reply:
Yes, it is. I changed the names to make it more clear.